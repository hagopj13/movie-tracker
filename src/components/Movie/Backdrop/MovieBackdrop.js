// @flow
import React from 'react';
import type { Node } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';

import configSelectors from 'store/config/config.selectors';

type Props = {
  backdropPath?: string,
  fullBackdropPath: string,
  children: Node,
};

const useStyles = makeStyles((theme) => ({
  backdrop: {
    boxShadow: 'inset 0 0 0 100vw rgba(0, 0, 0, 0.7)',
    backgroundImage: (props) => `url(${props.fullBackdropPath})`,
    backgroundSize: 'cover',
    backgroundPosition: 'center, center',
  },
  noImageBackdrop: {
    backgroundColor: theme.palette.grey[500],
  },
}));

export const MovieBackdrop = (props: Props) => {
  const { backdropPath, fullBackdropPath, children } = props;

  const classes = useStyles(props);

  const isImageFound = backdropPath && fullBackdropPath;

  return (
    <div className={isImageFound ? classes.backdrop : classes.noImageBackdrop}>{children}</div>
  );
};

const mapStateToProps = createStructuredSelector({
  fullBackdropPath: (state, ownProps: Props) =>
    configSelectors.createBackdropFullPathSelector(ownProps.backdropPath)(state),
});

MovieBackdrop.defaultProps = {
  backdropPath: null,
};

export default connect(mapStateToProps)(MovieBackdrop);
