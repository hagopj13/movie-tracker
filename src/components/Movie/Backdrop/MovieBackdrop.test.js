import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { MovieBackdrop } from './MovieBackdrop';

describe('MovieBackdrop component', () => {
  let shallow;
  let backdropPath;
  let fullBackdropPath;
  let wrapper;

  const ChildComponent = () => <div>Some child component</div>;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    backdropPath = 'pathToBackdropImage.png';
    fullBackdropPath = 'fullPathToBackdropImage.png';
    const mockProps = {
      backdropPath,
      fullBackdropPath,
    };
    wrapper = shallow(
      <MovieBackdrop {...mockProps}>
        <ChildComponent />
      </MovieBackdrop>,
    );
  });

  it('should render the MovieBackdrop component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });
});
