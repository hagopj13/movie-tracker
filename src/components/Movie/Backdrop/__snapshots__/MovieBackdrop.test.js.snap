// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBackdrop component should render the MovieBackdrop component correctly 1`] = `
<div
  className="makeStyles-backdrop-1 makeStyles-backdrop-3"
>
  <ChildComponent />
</div>
`;
